packages/typespec-client-generator-core/src/example.ts

```diff
@@ -0,0 +1,631 @@
+import {
+  Diagnostic,
+  DiagnosticCollector,
+  NoTarget,
+  SourceFile,
+  createDiagnosticCollector,
+  resolvePath,
+} from "@typespec/compiler";
+import { HttpStatusCodeRange } from "@typespec/http";
+import { resolveOperationId } from "@typespec/openapi";
+import {
+  SdkAnyExample,
+  SdkArrayExample,
+  SdkArrayType,
+  SdkBodyModelPropertyType,
+  SdkClientType,
+  SdkDictionaryExample,
+  SdkDictionaryType,
+  SdkHttpOperation,
+  SdkHttpOperationExample,
+  SdkHttpParameter,
+  SdkHttpParameterExample,
+  SdkHttpResponse,
+  SdkHttpResponseExample,
+  SdkModelExample,
+  SdkModelPropertyType,
+  SdkModelType,
+  SdkNullExample,
+  SdkServiceMethod,
+  SdkServiceOperation,
+  SdkType,
+  SdkTypeExample,
+  SdkUnionExample,
+  isSdkFloatKind,
+  isSdkIntKind,
+} from "./interfaces.js";
+import { TCGCContext, getValidApiVersion } from "./internal-utils.js";
+import { createDiagnostic } from "./lib.js";
+
+interface LoadedExample {
+  readonly relativePath: string;
+  readonly file: SourceFile;
+  readonly data: any;
+}
+
+/**
+ * Load all examples for a client
+ * 
+ * @param context 
+ * @param client 
+ * @returns a map of all operations' examples, key is operation's operation id, 
+ * value is a map of examples, key is example's title, value is example's details
+ */
+async function loadExamples<TServiceOperation extends SdkServiceOperation>(
+  context: TCGCContext,
+  client: SdkClientType<TServiceOperation>
```

Comment:
This method seems only use `apiVersion` from the `client`. It seems misleading to send a `client`.

Reply:
what do you mean misleading? it is not exported and only for internal logic.

Reply:
This is just a comment on dev for method, as e.g. a string can get the job done, why sending a whole client.
It kind of making the method hard to understand (what's it do with the client?).

Just a dev perspective about code readability.